src/dmd/globals.d

```diff
@@ -149,6 +149,7 @@ struct Param
                             // https://issues.dlang.org/show_bug.cgi?id=16997
     bool fixAliasThis;      // if the current scope has an alias this, check it before searching upper scopes
     bool vsafe;             // use enhanced @safe checking
+    bool cpp98;             // follow C++98 type system issues rather than C++11
```

Comment:
I think an enum should be used instead. It's fine to an an enum with only one member.

Reply:
A 2 state enum has no advantage over a bool.

Reply:
It's not 2 states. There's a whole lot of C++ versions, C++20 will create a new divide of the sort we're trying to deal with right here when they introduce `char8_t` for utf8 for instance.
I've observed STL structures change across these std versions too.